pkg/operation/botanist/controlplane.go

```diff
@@ -237,3 +244,32 @@ func waitUntilNoPodsExistAnymore(ctx context.Context, c client.Client, namespace
 	}
 	return flow.Parallel(fns...)(ctx)
 }
+
+func waitUnitlEtcdIsReady(ctx context.Context, seedClient client.Client, namespace string) error {
+	timeoutContext, cancel := context.WithTimeout(ctx, time.Minute*5)
+	defer cancel()
+
+	return retry.Until(timeoutContext, time.Second*5, func(ctx context.Context) (done bool, err error) {
+		etcdList := &druidv1alpha1.EtcdList{}
+		if err := seedClient.List(ctx, etcdList, client.InNamespace(namespace)); err != nil {
+			return retry.SevereError(err)
+		}
+
+		if meta.LenList(etcdList) == 0 {
+			return retry.Ok()
+		}
+
+		var notReadyETCDs []string
+		for _, etcd := range etcdList.Items {
+			if pointer.BoolDeref(etcd.Status.Ready, false) {
+				notReadyETCDs = append(notReadyETCDs, etcd.Name)
+			}
+		}
+
+		if len(notReadyETCDs) == 0 {
+			return retry.Ok()
+		}
+
+		return retry.MinorError(fmt.Errorf("etcds %s still not ready", notReadyETCDs))
+	})
+}
```

Comment:
Can't we simply call https://github.com/gardener/gardener/blob/498403ac7be3f528d51e3647882155414f22289e/pkg/operation/botanist/etcd.go#L148-L154

Reply:
This is nice. 👍